Load firmware via udev/fxload

pvachon opened the pull request:
This is a trivial change that removes the firmware load from the hands of the application.

Rather than having to have a built-in firmware loader in the app, load the firmware using fxload, which runs (if configured as in the example udev rules file) when the device is connected.

Verified with the Gendex II (GXS-700 "Large") - I do not have the small variant to test with. Firmware is the same as that which shipped with GxPicture 3.8.1.

pvachon committed:
Load firmware via udev/fxload

Rather than having to have a built-in firmware loader in the app,
instead load the firmware using fxload, which runs (if configured as in
the example udev rules file) when the device is connected.

JohnDMcMaster commented:
Thanks! I fiddled with this before but I couldn't get the multi-stage stuff to work correctly. Will test out in the next few days, and if all looks good will merge it in. There will also need to be a follow up to remove the .py code, but not a big deal for now since that code will just be ignored if its already loaded.

Also needs the Dexis rules to be complete. FYI the VID/PIDs can be found here: https://github.com/JohnDMcMaster/gxs700/blob/master/gxs700/fw.py

pvachon committed:
Add Dexis Platinum on spec

I don't have one of these sensors to test with, but based on what's in
the existing scripts it should work. Testing appreciated.

JohnDMcMaster commented:
Tested on large size, seems to work. Thanks!

EDIT: tested on all three
